src/servers/src/influxdb.rs

```diff
@@ -86,12 +93,34 @@ impl TryFrom<InfluxdbRequest> for RowInsertRequests {
             });
             row_writer::write_fields(table_data, fields, &mut one_row)?;
 
-            // timestamp
-            row_writer::write_ts_to_nanos(
+            // Align to the table's time index unit if needed.
+            let target_unit = if auto_align_precision {
+                catalog_manager
```

Comment:
I'm unsure if introducing the catalog here is a good idea.

Reply:
Agree. But I don't have a better solution. Maybe adds some handler to breaking the strong coupling, just like:

```rust
pub struct AlignPrecisionHandler {
   caatlog_manager: CatalogManagerRef,
}
impl AlignPrecisionHandler {
   pub async fn get_target_time_unit(query_ctx: &QueryContextRef, table_name: &str) -> Result<TimeUnit> {
        ....
   }
}

....
to_row_insert_requests(AlignPrecisionHandler{catalog_manager}, ...);


```